src/Cassandra/Requests/RequestExecution.cs

```diff
@@ -158,8 +158,18 @@ private void HandleRowSetResult(Response response)
         private void HandleSchemaChange(ResultResponse response)
         {
             var result = FillRowSet(new RowSet(), response);
-            //Wait for the schema change before returning the result
-            _parent.SetCompleted(result, () => _session.Cluster.Metadata.WaitForSchemaAgreement(_connection));
+
+            // This is a schema change so initialize schema in agreement as false
+            result.Info.SetSchemaInAgreement(false);
+
+            // Wait for the schema change before returning the result
+            _parent.SetCompleted(
+                result, 
+                () =>
+                {
+                    var schemaAgreed = _session.Cluster.Metadata.WaitForSchemaAgreement(_connection);
```

Comment:
We could improve this by using non-blocking code, but this is out of scope of this task. We can create a ticket to improve this by using non-blocking code, wdyt?

Reply:
I definitely agree but I prefer to create a ticket for that just because that is something that can affect (or even break) some other things and this task is just a new feature. 

If this wasn't one of my first tasks I probably would have changed it right away 😄 but I'd rather take it slowly.